### PR TITLE
Scripts/Zone: Update Dustwallow Marsh zone file

### PR DESCRIPTION
**Pull request by offl**
**Changes proposed:**

-  Move Nat Pagle to DB
-  Remove unused headers, outdated file info
-  Update spell scripts to modern standards, validate only spells used in spell scripts(only one is used)

**Target branch(es):**

- [x] 3.3.5
- [ ] master

**Issues addressed:**

none

**Tests performed:**

Builds, tested, works

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp [hunk 1]
```diff
@@ -15,215 +15,104 @@
...
-        SpellScript* GetSpellScript() const override
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        for (std::list<WorldObject*>::iterator itr = targets.begin(); itr != targets.end();)
         {
-            return new spell_ooze_zap_channel_end_SpellScript();
+            if ((*itr)->GetTypeId() == TYPEID_PLAYER && (*itr)->ToPlayer()->GetQuestStatus(GetSpellInfo()->Effects[EFFECT_1].CalcValue()) == QUEST_STATUS_INCOMPLETE)
+                ++itr;
+            else
+                targets.erase(itr++);
```
Comment (ccrs) [inline src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp:94]: erase returns the next pointer

https://www.cplusplus.com/reference/list/list/erase/

File: src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp [hunk 2]
```diff
@@ -15,215 +15,104 @@
...
-            void Register() override
-            {
-                OnEffectHitTarget += SpellEffectFn(spell_ooze_zap_channel_end_SpellScript::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
-            }
-        };
-
-        SpellScript* GetSpellScript() const override
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        for (std::list<WorldObject*>::iterator itr = targets.begin(); itr != targets.end();)
```
Comment (ccrs) [inline src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp:89]: you can use auto for iterators

File: src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp (hunk 1 above)
Comment (ccrs) [reply src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp:94]: itr = target.erase

**Comment by offl** [comment]
@ccrs these PRs are mostly about updating scripts to new model without additional changes

File: src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp (hunk 2 above)
Comment (Ovahlord) [reply src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp:89]: Lol, no. You have the type literally in front of your face and it's a very basic one so no auto when there is no real reason to use it (like  type<type<type<type>>> insanity. We aint using C# here lul. Also that's not part of this PR. He's switching the scripts over to the register macro model, so cutting away the script loaders basically.

Comment (ccrs) [reply src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp:89]: its the standard here, auto is allowed for iterators.
about this just being a refactor, yeah I missed this code actually already existed, my bad, lets move along.

**Pull request merged**